### PR TITLE
Upgrade code to 2020 style buiild PASS on node v6, v8, v10, v12, v14

### PR DESCRIPTION
**Pull request by UrielCh**
Hi,

Since I do not have any oscilloscope / working testing hardware, I can not test if may change work properly.

I would like to control some SK6812 LED strip directly from nodeJS, but I'm a newbie on this hardware part.

### CODE REVIEW

File: mailbox.c [hunk 1]
```diff
@@ -40,45 +40,40 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
-{
-   int mem_fd;
-   unsigned offset = base % PAGE_SIZE;
-   base = base - offset;
-   /* open /dev/mem */
-   if ((mem_fd = open("/dev/mem", O_RDWR|O_SYNC) ) < 0) {
-      printf("can't open /dev/mem\nThis program should be run as root. Try prefixing command with: sudo\n");
-      exit (-1);
+void *mapmem(unsigned base, unsigned size, const char *mem_dev) {
+   unsigned pagemask = ~0UL ^ (getpagesize() - 1);
```
Comment (xinkaiwang) [inline mailbox.c]: Where is this getpagesize() get defined? What is the benefit of changing PAGE_SIZE into a function?

File: mailbox.c [hunk 2]
```diff
@@ -40,45 +40,40 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
    }
-   void *mem = mmap(
-      0,
-      size,
-      PROT_READ|PROT_WRITE,
-      MAP_SHARED/*|MAP_FIXED*/,
-      mem_fd,
-      base);
+
+   void *mem = mmap(0, size, PROT_READ | PROT_WRITE, MAP_SHARED, mem_fd, base & pagemask);
```
Comment (xinkaiwang) [inline mailbox.c]: Is this a bug fix or defensive coding?
 (base) -> (base&pagemask).  
It's not very clear to me whether this change behavior or not. Inn general, I would avoid mixing formatting change with other kind of code change in the same PR.

Comment (xinkaiwang) [reply mailbox.c]: OK, got it, you prefer to use same `base` + 2 different mask (pagemask, offsetmask) to replace base + offset. 
personally I feel the old style is simpler for reader and less error prone.

**Review by xinkaiwang — CHANGES_REQUESTED** [review]
Thanks for your contribution. Over all looks great. 
Can you please change it back to base+offset then I will approve this PR?

File: mailbox.c (hunk 1 above)
Comment (UrielCh) [reply mailbox.c]: I took a lot a version of mailbox.c over github, to pick and merge the best part of the code.

**Comment by UrielCh** [comment]
extra Mask stuff removed.

**Comment by UrielCh** [comment]
yeeeh merged !!

**Pull request merged**